e2e/tests/wasm/upgrade_test.go

```diff
@@ -99,6 +99,13 @@ func (s *IBCWasmUpgradeTestSuite) UpgradeChain(ctx context.Context, chain *cosmo
 	err = testutil.WaitForBlocks(timeoutCtx, int(haltHeight-height)+1, chain)
 	s.Require().Error(err, "chain did not halt at halt height")
 
+	var allNodes []testutil.ChainHeighter
+	for _, node := range chain.Nodes() {
+		allNodes = append(allNodes, node)
+	}
+
+	testutil.WaitForInSync(ctx, chain, allNodes...)
```

Comment:
curious as to why other tests weren't running into issues because of this 🤔

Reply:
same! I can add this snippet to the other upgrade tests as well, I see no reason why we shouldn't have it 🤷🏻‍♂️

Reply:
In practise node operators will set their halt height or just wait for a node to panic in the upgrade module requesting the new chain binary - so anything that reduces any room for out of sync nodes in the test is good imo!